generate_figures/utilities.py

```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import wandb
+
+def plot_params(figsize, fontsize, ticksize):
+    params = {
+        "text.usetex": True,
+        "font.family": "serif",
+        "figure.figsize": figsize,
+        "figure.dpi": 80,
+        "figure.edgecolor": "k",
+        "font.size": fontsize,
+        "axes.labelsize": fontsize,
+        "axes.titlesize": fontsize,
+        "xtick.labelsize": ticksize,
+        "ytick.labelsize": ticksize,
+    }
+    return(params)
+
+
+
+def get_results_by_tag(project, tag, test_results = False):
+    api = wandb.Api()
+    count=0
+    runs = api.runs(project, filters={"tags": tag})
+    df_test_results = pd.DataFrame()
+    for i in range(len(runs)):
+        count+=1
+        df = pd.DataFrame()
+        cols = dict(runs[i].summary)
+        cols.update(dict(runs[i].config))
+        cols['id'] = runs[i].id
+        df = pd.DataFrame.from_dict(cols, orient='index').T
+ 
+        if test_results:
```

Comment:
It was breaking when I had if `test_results == True` where it would only run it the first time through.....this works but it makes me think that it's fragile.

Reply:
I am not sure exactly what the issue you are talking about is here?  Is it finicky and sometimes works/fails on the same run or are there some ones it deterministically breaks on and others it works great?

Reply:
@jlperla in its current state it always works. But if instead I had it as `test_results == True` it would only grab the first run. And I just don't really know why one works and not the other doesn't because I thought `if test_results == True` is equivalent to `if test_results:`